kotlin/offline-geocode/src/main/java/com/esri/arcgisruntime/sample/offlinegeocode/MainActivity.kt

```diff
@@ -0,0 +1,278 @@
+      // attributes from a click-based search
+      val street = geocode.attributes["Street"].toString()
+      val city = geocode.attributes["City"].toString()
+      val state = geocode.attributes["State"].toString()
+      val zip = geocode.attributes["ZIP"].toString()
+      val detail = "$city, $state $zip"
+      val address = "$street,$detail"
+      displayGeocodeResult(point, address)
+
+    } catch (e: Exception) {
+      val error = "Error getting geocode results: " + e.message
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+      Log.e(TAG, error)
+    }
+  }
+
+  /**
+   * Draw a point and open a callout showing geocode results on map.
+   *
+   * @param resultPoint geometry to show where the geocode result is
+   * @param address     to display in the associated callout
+   */
+  private fun displayGeocodeResult(
+    resultPoint: Point,
+    address: CharSequence
+  ) {
+    // dismiss the callout if showing
+    if (mapView.callout.isShowing) {
+      mapView.callout.dismiss()
+    }
+    val graphicsOverlay = mapView.graphicsOverlays[0]
+    // remove any previous graphics/search results
+    graphicsOverlay.graphics.clear()
+    // create graphic object for resulting location and add it to the ographics overlay
+    graphicsOverlay.graphics.add(Graphic(resultPoint, pointSymbol))
+    // zoom map to geocode result location
+    mapView.setViewpointAsync(Viewpoint(resultPoint, 8000.0), 3f)
+    showCallout(resultPoint, address)
+  }
+
+  /**
+   * Show a callout at the given point with the given text.
+   *
+   * @param point to define callout location
+   * @param text  to define callout content
+   */
+  private fun showCallout(point: Point, text: CharSequence) {
+    val calloutTextView = TextView(this).apply {
+      this.text = text
+    }
+    mapView.callout.apply {
+      location = point
+      content = calloutTextView
+    }
+    mapView.callout.show()
+  }
+
+  /**
+   * Sets up the address SearchView and uses MatrixCursor to show suggestions to the user
+   */
+  private fun setupAddressSearchView() {
+    // get the list of pre-made suggestions
+    val suggestions = resources.getStringArray(R.array.suggestion_items)
+
+    // set up parameters for searching with MatrixCursor
+    val columnNames = arrayOf(BaseColumns._ID, "address")
+    val suggestionsCursor = MatrixCursor(columnNames)
+
+    // add each address suggestion to a new row
+    suggestions.forEachIndexed { i, s -> suggestionsCursor.addRow(arrayOf(i, s)) }
+
+    // create the adapter for the search view's suggestions
+    searchView.apply {
+      suggestionsAdapter = SimpleCursorAdapter(
+        this@MainActivity,
+        R.layout.suggestion,
+        suggestionsCursor,
+        arrayOf("address"),
+        intArrayOf(R.id.suggestion_address),
+        0
+      )
+      // show the suggestions as soon as the user opens the search view
+      findViewById<AutoCompleteTextView>(R.id.search_src_text).threshold = 0
```

Comment:
I see that this doesn't work with kotlin synthetic. Any idea why?

Reply:
So this line is meant to fix the problem you mentioned above, which is that the suggestions should appear when the search view is opened. This behavior works for me, so I'm curious if it doesn't work for you

Reply:
The reason this does not use kotlin synthetic is because this view is not directly part of activity_main. Because this is in an `apply` block, this is really doing a `searchView.findViewById<...>(...)`. The `R.id.search_src_text` is an android-defined id, and under the hood, the searchView contains an `AutoCompleteTextView`. I can't remember what the more complicated solution for this behavior was, but I remember struggling for a long time trying to figure out how to implement it succinctly. Basically, this line is unpacking the searchView, grabbing the view that controls the suggestions, and setting its threshold for auto-completing to 0 so that the user doesn't have to type any characters to display suggestions